Constructive Code Review/outline.txt

```diff
@@ -0,0 +1,205 @@
+              the rest of this for a week. Someone else can feel free
+              to step in.” Don't feel like you married the patch.
+        - Stuff falls out of working memory pretty fast, and it's
+          expensive to load back in.
+        - It's energizing to get your patches turned around quickly.
+- Emotional issues
+    - Insecurity
+          a kind of fear
+        - Who in the room has ever secretly felt (or feared) they're
+          not as competent as the engineers around you?
+              (If they don't put up their hands: "Of course you don't.
+              It's a secret.")
+        - Almost everybody does. It doesn't really go away. Realize
+          nobody's judging you; they're too busy worrying about people
+          judging them.
+        - Realize you've already made it.
+            - If you're hired, congratulations. We don't have an easy
+              hiring process. You're competent to do this job.
+            - Sometimes somebody will find a problem in your PR because
+              they know something you don't. Sometimes it's just having
+              a fresh set of eyes.
+        - Who's afraid their imposter syndrome isn't as good as
+          everybody else's?
+        - It's been a help to me to notice when I'm feeling defensive
+          or angry.
+            - Why am I feeling that way? Sometimes that reveals things
+              I believe that aren't true or things I need to think
+              about.
+            - I generally make bad decisions and say things I regret
+              when feeling that way.
+            - It's a good time to talk a break and come back with a
+              clear head.
+    - Feeling short on time
+          There will always be more to review than can be reviewed. So
+          we have to learn to adapt to that reality: lower standards,
+          never sleep, or just go at a sustainable pace. It can do a
+          lot for a contrib's peace of mind just to know you've seen
+          that the patch is there and that you haven't forgotten about
+          it. Sometimes I even let people know "It's next in my queue."
+        - Consider: what's the risk if there's something wrong with it?
+            - Will it cause a lot of damage?
+            - Will it be easy to fix, or is there, say, a data-exchange
+              format being defined?
+    - Interpersonal issues
+        - When you're in person with your team, make a point of
+          establishing good relations with them.
+            - Then when you have to critique each other or talk on a
+              low-bandwidth medium like IRC, you'll know how to take
+              each other. You'll have some built-up offense-buffering.
+    - Other feelings
+        - When you can't decide what to say (something feels too bad to
+          merge), you can always fall back to articulating your
+          emotions: "I feel uncomfortable about leaving plain-text
+          passwords on disk."
```

Comment:
In this case I'd rather appeal to authority like "Our security policy/practices/etc. don't allow us to store passwords in plain-text, these need to be stored in (some way consistent with our security practices)", this way it isn't you imposing your opinion, but pointing out why something isn't acceptable. If the policy is inappropriate there should be a discussion about that separately from the patch.

We can't possibly (and shouldn't try to) have written policies about everything, if the reviewee pushes back on this point ("it doesn't matter that they are plain-test because xyz") I'd start a larger discussion, since for this example at least there should be a practice around it that everyone understands (even if they disagree or have reservations.)